src/ec/eddsa.rs

```diff
@@ -61,8 +105,11 @@ impl signature_impl::VerificationAlgorithmImpl for EdDSA {
 }
 
 
+#[allow(improper_ctypes)]
 extern {
-    #[cfg(test)]
+    fn ED25519_keypair(out_private_key: *mut u8/*[64]*/,
+                       rng: *mut RAND) -> c::int;
```

Comment:
Please use two `extern` blocks, one with `allow(improper_ctypes)` and one without. Only `E25519_keypair` should be marked `allow(improper_ctypes)`